lib/vacations.ex

```diff
@@ -0,0 +1,23 @@
+defmodule GlassFactoryApi.Vacations do
+  @moduledoc """
+
+  """
+
+  alias GlassFactoryApi.ApiClient
+  alias GlassFactoryApi.Vacations.Vacation
+
+  @doc """
+
+  """
+
+  def get_vacations(user_id \\ nil, config \\ %{}) do
+    with {:ok, %{status: 200, body: body}} <- ApiClient.get(vacations_path(user_id), config) do
+      {:ok, Enum.map(body, &Vacation.to_struct(&1))}
+    else
+      {:ok, %{status: 404}} -> {:error, "Can't find the vacations from user id #{user_id}"}
+    end
+  end
+
+  defp vacations_path(user_id) when (is_nil(user_id) or user_id == ""), do: "vacations"
```

Comment:
This guard clause was the solution I've found to solve this problem. Guys, can you collaborate with me to solve this problem from another perspective and improve me knowledge with the language? #gratidao

Reply:
why `user_id` can come as `nil` or `""`? 🤔

Reply:
It's optional in the query string params. So, if you want all vacations from all users, the user_id can be an empty string and if you want all vacations from an specific user, you need inform the user_id.

I've answered you my friend?

Reply:
Since it's an optional parameter like the others, we can instead accept a bag of params instead and keep it more flexible:

```elixir
def get_vacations(params \\ [], config \\ %{}) do
  ApiClient.get(vacations_path(user_id), params, config)
end

# That would require a tiny adaptation in the ApiClient itself, but I think it's worth it:
def get(resource, params \\ [], configuration) do
  config = Configuration.build(configuration)
  Tesla.get(tesla_client(config), resource, query: params)
end
```

Reply:
@britto Can you help me with the content of `params` variable? Is it a map or a string or something different? 😅

Reply:
I did search with @Cendrao and we've found that `params` variable is a list of map.

https://hexdocs.pm/tesla/Tesla.html#get/3

Reply:
> Can you help me with the content of params variable? Is it a map or a string or something different?

Sure! I'd love to.

> we've found that params variable is a list of map

In fact, the query params type is a "keyword list-like" thing, where keys can be strings, in addition to atoms: https://hexdocs.pm/tesla/Tesla.Env.html#t:query/0